clients/drcachesim/tools/view.cpp

```diff
@@ -118,37 +146,26 @@ view_t::process_memref(const memref_t &memref)
         std::cerr << "T" << memref.instr.tid << " ";
     }
 
-    if (memref.marker.type == TRACE_TYPE_MARKER) {
-        if (memref.marker.tid != 0 &&
-            printed_header_.find(memref.marker.tid) == printed_header_.end()) {
-            printed_header_.insert(memref.marker.tid);
+    if (memref.marker.type == TRACE_TYPE_MARKER ||
+        (knob_sim_refs_ > 0 && instr_count_ == knob_skip_refs_)) {
+        if (memref.instr.tid != 0 &&
```

Comment:
Why did we have to change `memref.marker.tid` to `memref.instr.tid`? I thought that we can use any `memref.*.tid` to get the tid as it has the same location in all structs in the `_memref_t` union.

Reply:
Just for consistency: the other .tid references in this code block are using instr.  Maybe keeping these and changing those others to marker is better though since the if() here will often match a marker -- making that change.